docs/reference/settings/data-frames-settings.asciidoc

```diff
@@ -0,0 +1,28 @@
+
+[role="xpack"]
+[[data-frames-settings]]
+=== Data frames settings in Elasticsearch
+++++
+<titleabbrev>{dataframe-transforms} settings</titleabbrev>
+++++
+
+You do not need to configure any settings to use {dataframe-transforms}. It is enabled by default.
+
+All of these settings can be added to the `elasticsearch.yml` configuration file. 
+The dynamic settings can also be updated across a cluster with the 
+<<cluster-update-settings,cluster update settings API>>.
+
+TIP: Dynamic settings take precedence over settings in the `elasticsearch.yml` 
+file.
+
+[float]
+[[general-data-frames-settings]]
+==== General {dataframe-transforms} settings
+
+`xpack.data_frames.num_transform_failure_retries`::
```

Comment:
I think `data_frames` should be changed to the singular `data_frame`.

There is already a data frame setting, and it begins with `xpack.data_frame` (singular): https://github.com/elastic/elasticsearch/blob/623a5ddac092c3443290580f6b145559e05d64f0/x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/XPackSettings.java#L44

(Also, is `xpack.ml.enabled` documented on the ML settings page? If so then maybe `xpack.data_frame.enabled` should be added to this page.)